beacon_node/http_api/src/attestation_rewards.rs

```diff
@@ -0,0 +1,260 @@
+
+    let state_slot = (epoch + 1).end_slot(T::EthSpec::slots_per_epoch());
+
+    let state_root = chain
+        .state_root_at_slot(state_slot)
+        .map_err(warp_utils::reject::beacon_chain_error)?
+        .ok_or_else(|| warp_utils::reject::custom_not_found("State root not found".to_owned()))?;
+
+    let mut state = chain
+        .get_state(&state_root, Some(state_slot))
+        .map_err(warp_utils::reject::beacon_chain_error)?
+        .ok_or_else(|| warp_utils::reject::custom_not_found("State not found".to_owned()))?;
+
+    //--- Calculate ideal_rewards ---//
+    let participation_cache = ParticipationCache::new(&state, spec)
+        .map_err(|e| custom_not_found(format!("Unable to get participation_cache! {:?}", e)))?;
+
+    let previous_epoch = state.previous_epoch();
+
+    let mut ideal_rewards_hashmap = HashMap::new();
+
+    let flag_index = 0;
+    let weight = 0;
+    let base_reward = 0;
+
+    for flag_index in [
+        TIMELY_SOURCE_FLAG_INDEX,
+        TIMELY_TARGET_FLAG_INDEX,
+        TIMELY_HEAD_FLAG_INDEX,
+    ]
+    .iter()
+    {
+        let weight = get_flag_weight(*flag_index)
+            .map_err(|e| custom_not_found(format!("Unable to get weight! {:?}", e)))?;
+
+        let unslashed_participating_indices = participation_cache
+            .get_unslashed_participating_indices(*flag_index, previous_epoch)
+            .map_err(|e| {
+                custom_not_found(format!(
+                    "Unable to get unslashed_participating_indices! {:?}",
+                    e
+                ))
+            })?;
+
+        let unslashed_participating_balance = unslashed_participating_indices
+            .total_balance()
+            .map_err(|e| {
+                custom_not_found(format!(
+                    "Unable to get unslashed_participating_balance! {:?}",
+                    e
+                ))
+            })?;
+
+        let unslashed_participating_increments = unslashed_participating_balance
+            .safe_div(spec.effective_balance_increment)
+            .map_err(|e| {
+                custom_not_found(format!(
+                    "Unable to get unslashed_participating_increments! {:?}",
+                    e
+                ))
+            })?;
+
+        let total_active_balance = participation_cache.current_epoch_total_active_balance();
+
+        let active_increments = total_active_balance
+            .safe_div(spec.effective_balance_increment)
+            .map_err(|e| custom_not_found(format!("Unable to get active_increments! {:?}", e)))?;
+
+        let base_reward_per_increment = BaseRewardPerIncrement::new(total_active_balance, spec)
+            .map_err(|e| {
+                custom_not_found(format!("Unable to get base_reward_per_increment! {:?}", e))
+            })?;
+
+        for effective_balance_eth in 0..=32 {
+            let base_reward = get_base_reward(
+                &state,
+                effective_balance_eth,
```

Comment:
This second argument is actually a _validator index_, not an effective balance.

We either need to modify the `get_base_reward` function in `consensus` (potentially risky) or duplicate some of its logic here, like so:

```rust
let base_reward = effective_balance_eth.safe_mul(base_reward_per_increment.as_u64())?;
```